python/cudf/cudf/utils/dtypes.py

```diff
@@ -425,11 +476,73 @@ def dtype_to_pylibcudf_type(dtype) -> plc.DataType:
     # libcudf types don't support timezones so convert to the base type
     elif isinstance(dtype, pd.DatetimeTZDtype):
         dtype = _get_base_dtype(dtype)
+    elif isinstance(dtype, pd.StringDtype):
+        dtype = CUDF_STRING_DTYPE
     else:
-        dtype = np.dtype(dtype)
+        dtype = pandas_dtypes_to_np_dtypes.get(dtype, dtype)
+        try:
+            dtype = np.dtype(dtype)
+        except TypeError:
+            dtype = cudf.dtype(dtype)
```

Comment:
When do we hit the `TypeError` here? (I see the same pattern below)

For context, it would be great to avoid having to call `cudf.dtype` here for the circular dependency